### ISSUE TITLE
Option to remove contributors from a package with `trident rectify`, or automatic overwriting of Josiah Carberry when new contributors are added.

### ISSUE THREAD

**Issue by TCLamnidis** [poseidon-framework/poseidon-hs#289]
Creating a new package semi-automatically is currently hampered by Josiah Carberry. in the contributor field.
I would like a way to create/update the contributor field to remove Josiah (and potentially, though less importantly, other contributors) from the list.

I would suggest either a `--removeContributor` option to `trident rectify`, to contrast the existing `--newContributor` option,
or a special case being added for Josiah Carberry, where he is automatically removed from the contributor list when another contributor is added to the package.

This would minimise the amount of manual editing of the POSEIDON.yml that a user has to do to finalise a package. That said, I can see that this is only an issue in cases where one user is creating many packages at a time, so maybe not a priority.

**Comment by stschiff** [comment]
We already have a `--minimal` flag. How about we make it so that when that is chosen, no dummy values are filled in the YAML file?

**Comment by TCLamnidis** [comment]
`--minimal` also creates no `janno`, or `bib` file, just a `POSEIDON.yml` to go with the genotypes.

**Comment by stschiff** [comment]
Hmm, I'm concerned that we should think hard before adding more CLI options here for very specific YAML changes. The usual rectify-workflow is a quite specific edit of the YAML, geared towards "typical" update workflows. If you want custom changes to the YAML, you could just do that directly. For example, there are tools like https://github.com/andywerner/yaml_cli that you can check, which allow you to edit YAML from the CLI.

Or in fact you could do this:

```
python3 -c 'import yaml; d=yaml.load(open("POSEIDON.yml", "r").read(), Loader=yaml.CLoader); d["contributor"] = {"name": "Thiseas", "email" : "Thiseas.mail.com"};open("POSEIDON2.yml", "w").write(yaml.dump(d))'
```

**Comment by TCLamnidis** [comment]
I would argue that removing Josiah Carberry from the contributors is very much a typical update workflow.
One could even argue that validation of a package for addition to the PCA _SHOULD_ check for his absence from the contributors.
I understand that adding a contributor with an update to a Poseidon package is more likely than removing one, and that removal of contributors should perhaps be less easy to do on purpose (as it is a heftier decision to make anyway), but in that case I would still favour a special-case for Josiah be made. The way I see it, there is little value to an easy-to-use way to add a contributor to a package if one then HAS to edit the POSEIDON.yml by hand afterwards anyway. At that point, we could remove the Contributor addition altogether and rely on users editing manually, or coming up with their own python yaml editing solutions for that operation as well.
To be clear, I am obviously not in favour of foregoing this QOL option altogether. But since most packages will only be made once, the usefulness of this option is unclear, unless one doesn't need to also update the YAML file manually.

**Comment by stschiff** [comment]
OK, I think I understand that our choice of adding a dummy contributor is annoying since it nessitates a forced manual update.

The minimal solution would be to simply remove creation of a dummy contributor in `trident init`, as you have alluded to, @TCLamnidis. I think the reason @nevrome implemented it that way was to give users a pointer to not forget about the contributor. But we could perhaps mitigate that by adapting `trident validate` to output a warning if a package has no contributor.

A more involved suggestion would be to introduce Poseidon environment variables. For example, we could read from `POSEIDON_USER_NAME` and `POSEIDON_USER_EMAIL`, which can be set by a user. If `trident init` then finds these, it will use them as the first contributor, if not then the contributor field stays empty. I think this wouldn't affect rectify or validate, the variables only affect `init`.

What do you think?

**Comment by nevrome** [comment]
Here are the options I saw mentioned now:

1. `rectify` gets a `--removeContributor` option
2. `rectify` automatically removes Josiah when `--newContributor` is set
3. Change nothing and leave users and developers to their own device
4. remove Josiah from the `init` output (throw a warning with `validate` if the contributor is missing)
5. replace Josiah with a contributor read from environment variables

I think I'm with Thiseas that 3. is not ideal. 5. is not something I would like to do, because I dislike environment variable based configuration. 2. would surprise me as a user, because it's an unexpected side effect.

For me the decision is about 1. and 4. Between the two I think it really depends on how valuable Josiah is as a didactic tool. He's certainly a bit of an odd quirk of trident and maybe it's better to remove him. The `validate` warning would be a good alternative.

**Comment by stschiff** [comment]
Yes, I think I'm siding with option 4! This gives us a clear angle for implementation!

**Comment by nevrome** [comment]
Implemented and released in v1.5.0.0 :+1:
